### ISSUE TITLE
Add equality and comparison operators, pleeeease

### ISSUE THREAD

**Issue by dsyme** [gregoryyoung/nothing#1]

**Comment by gregoryyoung** [comment]
:)

**Comment by pezipink** [comment]
I agree.  Without said enhancement, this otherwise promising library is worth Nothing.

**Comment by mattjohnsonpint** [comment]
+1.  How else could I determine that I have Nothing?

**Comment by mattjohnsonpint** [comment]
Comparison is particularly interesting.  I mean, I think I know what it means to have nothing, or have more than nothing, but what is less than nothing?

**Comment by vossccp** [comment]
Wouldnt implementing Equals imply that „Nothing“ has some kind of a value? Wouldn’t this make “Nothing” “Something” and hence break the specs?
I believe Nothing.Instance==Nothing.Instance should always be false. There is an infinite number of Nothings out there. My Nothing is not your Nothing. (This also questions the design decision of having something like Nothing.Instance)

**Comment by marcingolenia** [comment]
Is there any progress on this? Maybe we can use records from C# 9?
